### PR TITLE
[System Tests] exclude unit test from system tests execution

### LINKED ISSUE #1584: [System Tests] Avoid unit tests execution when running system tests
**Describe the bug**
When running system tests we currently run unit tests (located in system tests package) and a report is provided that can disturb the actual reports for the system tests.

**Expected behavior**
Avoid the unit test execution when running system tests

### PR DESCRIPTION
**Pull request by franvila**
### Type of change

- Bugfix

### Description

Fixing the execution of unit tests for system test package when we don't need them.

This error came when I tried to avoid running `NonJVMClientsST` more than once when running tests for all Kafka clients.
```bash
mvn -q verify -DskipTests=true -DskipSTs=false '-Dtest=!io.kroxylicious.systemtests.NonJVMClientsST'
```

I have followed the same approach that Strimzi has using `Tag` to classify the tests in a way to exclude them from the system test execution.

### Additional Context

Fixes #1584 

### Checklist

- [x] Write tests
- [x] Update documentation
- [x] Make sure all unit/integration tests pass
- [x] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [x] If applicable to the change, [trigger the system test suite](../blob/main/DEV_GUIDE.md#jenkins-pipeline-for-system-tests).  Make sure tests pass.
- [x] If applicable to the change, [trigger the performance test suite](../blob/main/PERFORMANCE.md#jenkins-pipeline-for-performance). Ensure that any degradations to performance numbers are understood and justified.
- [x] Ensure the PR references relevant issue(s) so they are closed on merging.
- [x] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

> **_NOTE:_**  You must be a member of `@kroxylicious/developers` to trigger the system test and performance test suites.  If you are not part of this group, comment on the PR requesting a trigger, tagging `@kroxylicious/developers`.

### CODE REVIEW

**Comment by franvila** [comment]
@strimzi-ci run system tests

File: kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/TestTags.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+public final class TestTags {
+
+    private TestTags() {
+    }
+
+    /**
+     * Tag for unit tests located in system tests package.
+     */
+    public static final String UNIT = "unit";
```
Comment (SamBarker) [inline kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/TestTags.java:17]: I think if we are going this way we should add tags for system tests too.

File: kroxylicious-systemtests/pom.xml [hunk 2]
```diff
@@ -186,6 +186,7 @@
                                 <include>**/ST*.java</include>
                                 <include>**/*ST.java</include>
                             </includes>
+                            <excludedGroups>unit</excludedGroups>
```
Comment (SamBarker) [inline kroxylicious-systemtests/pom.xml:189]: Why does `ContainerTemplatesTest` get picked up at all? It doesn't match the `include` entries...

**Review by SamBarker — COMMENTED** [review]
I'm not opposed to tagging, but if we are going down that line I think we should tag all the tests. Then we could just run the `System` group rather than trying to intersect `**ST*.java && ! UNIT`

Before we go down that line I'd like to understand why the include globs aren't doing what we wanted.

File: kroxylicious-systemtests/pom.xml (hunk 2 above)
Comment (robobario) [reply kroxylicious-systemtests/pom.xml:189]: it's the `-Dtests`, overriding the includes

> Specify this parameter to run individual tests by file name, overriding the parameter includes and excludes. Each pattern you specify here will be used to create an include pattern formatted like **/${test}.java, so you can just type -Dtest=MyTest to run a single test called "foo/MyTest.java". The test patterns prefixed with a ! will be excluded.

**Comment by robobario** [comment]
Would`'-Dsurefire.excludes=**/NonJVMClientsST.java'` work instead? Running with `-X` that shows:

```
[DEBUG] Resolved included and excluded patterns: **/ST*.java, **/*ST.java, !**/NonJVMClientsST.java
```

**Comment by franvila** [comment]
> Would`'-Dsurefire.excludes=**/NonJVMClientsST.java'` work instead? Running with `-X` that shows:
> 
> ```
> mvn verify -DskipTests=true -DskipSTs=false '-Dsurefire.excludes=**/NonJVMClientsST.java' -X | grep 'Resolved included'
> [DEBUG] Resolved included and excluded patterns: **/ST*.java, **/*ST.java, !**/NonJVMClientsST.java
> ```
> 
> vs
> 
> ```
> mvn verify -DskipTests=true -DskipSTs=false '-Dtest=!io.kroxylicious.systemtests.NonJVMClientsST' -X | grep 'Resolved included'
> [DEBUG] Resolved included and excluded patterns: !io.kroxylicious.systemtests.NonJVMClientsST
> ```

Thanks! I didn't found something like that and I was very upset when knowing that `-Dtest` was overriding the includes.

I will test that and if we can avoid this PR I will close it. I don't like the idea to include `Tags` at this time, but it was the only way I found.

**Comment by franvila** [comment]
I have rethought the way to avoid running `NonJVMClientsST` test suite more times than needed. I have tagged it with `KAFKA_CLIENT_INDEPENDENT` tag that may be used by another test suites in the future if needed without the need to modify the command to exclude individual test suites rather than using `-DexcludeGroups` to exclude all the test suites that need to be excluded.

**Comment by franvila** [comment]
@strimzi-ci run system tests

File: kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java [hunk 3]
```diff
@@ -27,13 +28,15 @@
...
 
+import static io.kroxylicious.systemtests.TestTags.KAFKA_CLIENT_INDEPENDENT;
 import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * The non-JVM clients system tests.
  */
 @ExtendWith(KroxyliciousExtension.class)
+@Tag(KAFKA_CLIENT_INDEPENDENT)
```
Comment (SamBarker) [inline kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java:39]: The approach makes sense to me but I'm not sure about the naming! As the test is not independent of the client used. I think your aiming for a tag that identifies tests which don't use the Kafka Client from the class path. So maybe it should be `EXTERNAL_CLIENT` or `EXTERNAL_KAFKA_CLIENTS`.

**Review by SamBarker — APPROVED** [review]
LGTM - Well I have doubts about the naming but the approach works for me.

>Do you think we should add SystemTest tag to set by default includedGroups all the tests and then include/exclude the groups needed for each pipeline run?

No, well I don't see the need for it. I suggested it previously as the logical complement of the suggested `UNIT` tag. However I think the current proposed is a way of identifying a sub group of tests rather than an arbitrary breakdown. (I'm not entirely convinced I'm being logical about this so feel free to disagree).

File: kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java (hunk 3 above)
Comment (franvila) [reply kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java:39]: Yeah, I'm so bad on naming! I wanted to reflect that these tests ignore the `KAFKA_CLIENT` env variable.

Comment (SamBarker) [reply kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java]: Haha we all suck at naming at your playing on hard doing in a second language! Keep up the good work!

**Pull request merged**